example/lib/main.dart

```diff
@@ -33,58 +37,192 @@ class MyHomePage extends StatefulWidget {
+      final availableIcons = await MobileIconSwitcher.getAvailableIcons();
+      
+      setState(() {
+        _availableIcons = availableIcons;
+        _statusMessage = 'Current icon: ${currentIcon ?? 'Unknown'}';
+      });
+    } catch (e) {
+      setState(() {
+        _statusMessage = 'Error loading icon info: $e';
+      });
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         backgroundColor: Theme.of(context).colorScheme.inversePrimary,
         title: Text(widget.title),
       ),
-      body: Center(
+      body: Padding(
+        padding: const EdgeInsets.all(16.0),
         child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.stretch,
           children: <Widget>[
-            const Text(
-              'Change the Apps Icon:',
+            Card(
+              child: Padding(
+                padding: const EdgeInsets.all(16.0),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Text(
+                      'Status',
+                      style: Theme.of(context).textTheme.headlineSmall,
+                    ),
+                    const SizedBox(height: 8),
+                    Text(_statusMessage),
+                    const SizedBox(height: 8),
+                    Text('Platform supported: ${MobileIconSwitcher.isSupported}'),
+                  ],
+                ),
+              ),
+            ),
+            const SizedBox(height: 16),
+            Text(
+              'Modern API (Recommended)',
+              style: Theme.of(context).textTheme.headlineSmall,
+            ),
+            const SizedBox(height: 8),
+            ElevatedButton(
+              onPressed: () => _switchAppIconModern("first"),
+              child: const Text("Switch to First Icon"),
+            ),
+            ElevatedButton(
+              onPressed: () => _switchAppIconModern("second"),
+              child: const Text("Switch to Second Icon"),
             ),
             ElevatedButton(
-              onPressed: () {
-                switchAppIcon("first");
-              },
-              child: const Text("First Icon"),
+              onPressed: _resetAppIconModern,
+              child: const Text("Reset to Default Icon"),
+            ),
+            const SizedBox(height: 16),
+            Text(
+              'Legacy API (Deprecated)',
+              style: Theme.of(context).textTheme.headlineSmall,
             ),
+            const SizedBox(height: 8),
             ElevatedButton(
-              onPressed: () {
-                switchAppIcon("second");
-              },
-              child: const Text("Second Icon"),
+              onPressed: () => _switchAppIconLegacy("first"),
+              child: const Text("Legacy: Switch Icon"),
             ),
```

Comment:
Please also add the second icon for the legacy API
```suggestion
            ElevatedButton(
              onPressed: () => _switchAppIconLegacy("first"),
              child: const Text("Legacy: Switch to First Icon"),
            ),
            const SizedBox(height: 8),
            ElevatedButton(
              onPressed: () => _switchAppIconLegacy("second"),
              child: const Text("Legacy: Switch to Second Icon"),
            ),
```